File: java_bank/Main_no_comments.java
import java.util.Scanner;

public class Main_no_comments {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        Account acc = new Account("홍길동", 10000);
        System.out.printf("[%s]님의 계좌가 생성되었습니다. (초기 잔액: %,d원)%n", acc.getOwner(), acc.getBalance());

        boolean running = true;

        while (running) {
            System.out.println("\n=== 은행 메뉴 ===");
            System.out.println("1. 입금 | 2. 출금 | 3. 잔액 조회 | 4. 종료");
            System.out.print("선택 >> ");
            String choice = scanner.nextLine().trim();

            switch (choice) {
                case "1":
                    System.out.print("입금할 금액: ");
                    try {
                        long amount = Long.parseLong(scanner.nextLine().trim());
                        acc.deposit(amount);
                    } catch (NumberFormatException e) {
                        System.out.println("올바른 숫자를 입력하세요.");
                    }
                    break;

                case "2":
                    System.out.print("출금할 금액: ");
                    try {
                        long amount = Long.parseLong(scanner.nextLine().trim());
                        acc.withdraw(amount);
                    } catch (NumberFormatException e) {
                        System.out.println("올바른 숫자를 입력하세요.");
                    }
                    break;

                case "3":
                    System.out.printf("[%s]님의 현재 잔액: %,d원%n", acc.getOwner(), acc.getBalance());
                    break;

                case "4":
                    System.out.println("프로그램을 종료합니다.");
                    running = false;
                    break;

                default:
                    System.out.println("잘못된 선택입니다. 다시 입력해주세요.");
                    break;
            }
        }
        scanner.close();
    }
}

File: cpp_bank/main_no_comments.cpp
#include "Account_no_comments.h"
#include <iostream>
#include <string>
#include <windows.h>

int main() {
    SetConsoleOutputCP(CP_UTF8);

    Account acc("홍길동", 10000);

    Account acc2("김철수", 5000);

    std::cout << "[" << acc.getOwner() << "]님의 계좌가 생성되었습니다. (초기 잔액: "
              << formatWithComma(acc.getBalance()) << "원)" << std::endl;
    std::cout << "[" << acc2.getOwner() << "]님의 계좌가 생성되었습니다. (초기 잔액: "
              << formatWithComma(acc2.getBalance()) << "원)" << std::endl;

    bool running = true;
    std::string line;

    while (running) {
        std::cout << "\n=== 은행 메뉴 ===" << std::endl;
        std::cout << "1. 입금 | 2. 출금 | 3. 잔액 조회 | 4. 이체(friend 함수) | 5. 종료" << std::endl;
        std::cout << "선택 >> ";
        std::getline(std::cin, line);

        if (line == "1") {
            std::cout << "입금할 금액: ";
            std::getline(std::cin, line);
            try {
                long long amount = std::stoll(line);
                acc.deposit(amount);
            } catch (const std::exception&) {
                std::cout << "올바른 숫자를 입력하세요." << std::endl;
            }
        } else if (line == "2") {
            std::cout << "출금할 금액: ";
            std::getline(std::cin, line);
            try {
                long long amount = std::stoll(line);
                acc.withdraw(amount);
            } catch (const std::exception&) {
                std::cout << "올바른 숫자를 입력하세요." << std::endl;
            }
        } else if (line == "3") {
            std::cout << "[" << acc.getOwner() << "]님의 현재 잔액: "
                      << formatWithComma(acc.getBalance()) << "원" << std::endl;
            std::cout << "[" << acc2.getOwner() << "]님의 현재 잔액: "
                      << formatWithComma(acc2.getBalance()) << "원" << std::endl;
        } else if (line == "4") {
            std::cout << "이체할 금액 ([" << acc.getOwner() << "] -> ["
                      << acc2.getOwner() << "]): ";
            std::getline(std::cin, line);
            try {
                long long amount = std::stoll(line);
                transfer(acc, acc2, amount);
            } catch (const std::exception&) {
                std::cout << "올바른 숫자를 입력하세요." << std::endl;
            }
        } else if (line == "5") {
            std::cout << "프로그램을 종료합니다." << std::endl;
            running = false;
        } else {
            std::cout << "잘못된 선택입니다. 다시 입력해주세요." << std::endl;
        }
    }

    return 0;
}
